### PR TITLE
planner: fix "can't find column" when projection wrongly added above index lookup reader after agg pushed down

### LINKED ISSUE #33237: expression index and subquery return error:Can't find column plus(test.t2.c_int, 1) in schema Column: [test.t2.c_decimal] Unique key: []
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

drop table if exists t1, t2;
create table t1  (c_int int, c_decimal decimal(12, 6), primary key (c_int) nonclustered,key((c_int + 1))) ;
create table t2  like t1 ;
select * from t1 where c_decimal in (select c_decimal from t2 where t2.c_int + 1 = 8 + 1) ; 

### 2. What did you expect to see? (Required)

query return success.

### 3. What did you see instead (Required)

mysql> select * from t1 where c_decimal in (select c_decimal from t2 where t2.c_int + 1 = 8 + 1) ; 
ERROR 1105 (HY000): Can't find column plus(test.t2.c_int, 1) in schema Column: [test.t2.c_decimal] Unique key: []

[2022/03/18 09:02:48.471 +00:00] [WARN] [session.go:1867] ["compile SQL failed"] [conn=31573] [error="Can't find column plus(test.t2.c_int, 1) in schema Column: [test.t2.c_decimal] Unique key: []"] [SQL="select * from t1 where c_decimal in (select c_decimal from t2 where t2.c_int + 1 = 8 + 1)"]
[2022/03/18 09:02:48.471 +00:00] [INFO] [conn.go:1123] ["command dispatched failed"] [conn=31573] [connInfo="id:31573, addr:127.0.0.1:43874 status:10, collation:utf8_general_ci, user:root"] [command=Query] [status="inTxn:0, autocommit:1"] [sql="select * from t1 where c_decimal in (select c_decimal from t2 where t2.c_int + 1 = 8 + 1)"] [txn_mode=PESSIMISTIC] [timestamp=0] [err="Can't find column plus(test.t2.c_int, 1) in schema Column: [test.t2.c_decimal] Unique key: []\ngithub.com/pingcap/tidb/expression.(*Column).resolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/expression/column.go:501\ngithub.com/pingcap/tidb/expression.(*Column).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/expression/column.go:494\ngithub.com/pingcap/tidb/planner/core.(*PhysicalProjection).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:31\ngithub.com/pingcap/tidb/planner/core.(*basePhysicalPlan).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:675\ngithub.com/pingcap/tidb/planner/core.(*physicalSchemaProducer).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:664\ngithub.com/pingcap/tidb/planner/core.(*basePhysicalAgg).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:388\ngithub.com/pingcap/tidb/planner/core.(*basePhysicalPlan).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:675\ngithub.com/pingcap/tidb/planner/core.(*physicalSchemaProducer).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:664\ngithub.com/pingcap/tidb/planner/core.(*PhysicalHashJoin).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:79\ngithub.com/pingcap/tidb/planner/core.(*basePhysicalPlan).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:675\ngithub.com/pingcap/tidb/planner/core.(*physicalSchemaProducer).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:664\ngithub.com/pingcap/tidb/planner/core.(*PhysicalProjection).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:26\ngithub.com/pingcap/tidb/planner/core.(*basePhysicalPlan).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:675\ngithub.com/pingcap/tidb/planner/core.(*physicalSchemaProducer).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:664\ngithub.com/pingcap/tidb/planner/core.(*PhysicalProjection).ResolveIndices\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/resolve_indices.go:26\ngithub.com/pingcap/tidb/planner/core.physicalOptimize\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/optimizer.go:503\ngithub.com/pingcap/tidb/planner/core.DoOptimize\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/core/optimizer.go:285\ngithub.com/pingcap/tidb/planner.optimize\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/optimize.go:391\ngithub.com/pingcap/tidb/planner.Optimize\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/planner/optimize.go:222\ngithub.com/pingcap/tidb/executor.(*Compiler).Compile\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/executor/compiler.go:77\ngithub.com/pingcap/tidb/session.(*session).ExecuteStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/session/session.go:1860\ngithub.com/pingcap/tidb/server.(*TiDBContext).ExecuteStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/driver_tidb.go:223\ngithub.com/pingcap/tidb/server.(*clientConn).handleStmt\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1981\ngithub.com/pingcap/tidb/server.(*clientConn).handleQuery\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1850\ngithub.com/pingcap/tidb/server.(*clientConn).dispatch\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1345\ngithub.com/pingcap/tidb/server.(*clientConn).Run\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/conn.go:1095\ngithub.com/pingcap/tidb/server.(*Server).onConn\n\t/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/server/server.go:551\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1371"]
### 4. What is your TiDB version? (Required)

Release Version: v6.0.0-alpha
Edition: Community
Git Commit Hash: 9fcc6b090b02c8351aba4ee8a7f072e0e0c840e0
Git Branch: heads/refs/tags/v6.0.0-alpha
UTC Build Time: 2022-03-13 14:56:23
GoVersion: go1.16.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false

### PR DESCRIPTION
**Pull request by time-and-fate**
### What problem does this PR solve?

Issue Number: close #33237

Problem Summary:

Very similar to #23804, a projection was added between the partial agg and the final agg, and the schema between them was broken.
But #23804 is for `PhysicalTableReader`, this case is `PhysicalIndexLookUpReader`.

### What is changed and how it works?

Don't inject the extra projection if the agg was pushed down.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: planner/core/integration_test.go [hunk 1]
```diff
@@ -6243,3 +6243,14 @@ func TestTiFlashPartitionTableScan(t *testing.T) {
 	tk.MustExec("drop table rp_t;")
 	tk.MustExec("drop table hp_t;")
 }
+
+func TestIssue33237(t *testing.T) {
```
Comment (wjhuang2016) [inline planner/core/integration_test.go:6247]: We can push these test to explain test.

**Review by winoros — APPROVED** [review]
This is a case that can be solved after projection pushdown with full built-in function support.
Now we have projection soon, but when would we have full built-in function support?

File: planner/core/integration_test.go (hunk 1 above)
Comment (time-and-fate) [reply planner/core/integration_test.go]: Updated. Moved it to `explain.test`.

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**